media/whip_connection.go

```diff
@@ -77,13 +78,23 @@ type WHIPPeerConnection interface {
 	GetStats() webrtc.StatsReport
 }
 
+type PeerConnStats struct {
+	ID              string
+	BytesReceived   uint64
+	BytesSent       uint64
+	PacketsReceived uint32
+	PacketsSent     uint32
+}
+
 type TrackStats struct {
-	Kind webrtc.RTPCodecType
-	*stats.Stats
+	Kind        webrtc.RTPCodecType
+	Jitter      float64
+	PacketsLost int64
+	RTT         time.Duration
 }
 
 type MediaStats struct {
-	PeerConnStats webrtc.StatsReport
+	PeerConnStats PeerConnStats
```

Comment:
Do we have to make everything explicit because serialization had a hard time with the embedded `*stats.Stats` or was there another reason? It seems nice to be able to send _everything_ over if possible, without having to manually add in some other statistic we might want later.

Reply:
Yeah I tried serializing the track stats as json and most of the fields seem to be missing:

```
 {
  "LastPacketReceivedTimestamp": "2025-04-28T18:23:51.481138+01:00",
  "HeaderBytesReceived": 15672,
  "BytesReceived": 19590,
  "HeaderBytesSent": 0,
  "FractionLost": 0,
  "RemoteTimeStamp": "0001-01-01T00:00:00Z",
  "ReportsSent": 0
}
```
It might just be because there aren't `json` mappings on the types: https://github.com/pion/interceptor/blob/master/pkg/stats/received_stats.go#L28

There are also a lot of fields which aren't populated, they would just be empty values so feels a bit messy to print and send them all.

Reply:
Yeah don't necessarily have to log more than what we already do, but sending them over the wire to kafka seems harmless (unless we are concerned about the increase in size / volume of data) since you never know when some random field might end up being retrospectively useful. Anyway don't feel too strongly about this so whatever works.